### PR TITLE
App developer can restart an app with downtime

### PR DESCRIPTION
**Pull request by akrishna90**
## Is there a related GitHub Issue?
#170 

## What is this change about?
- adds POST v3/apps/:guid/actions/restart endpoint
 - implementation uses a polling mechanism to achieve restart synchronously

## Does this PR introduce a breaking change?
No

## Acceptance Steps
Follow acceptance on #170 

## Tag your pair, your PM, and/or team
@gnovv

### CODE REVIEW

**Review by gnovv — APPROVED** [review]
pairing with @akrishna90  11/17, lgtm

File: api/apis/app_handler_test.go [hunk 1]
```diff
@@ -2156,6 +2159,308 @@ var _ = Describe("AppHandler", func() {
 			})
 		})
 	})
+
+	Describe("the GET /v3/apps/:guid/actions/restart", func() {
```
Comment (tcdowney) [inline api/apis/app_handler_test.go:2163]: Nit: 

I think missing the word "endpoint" at the end 😛

File: api/apis/app_handler_test.go [hunk 2]
```diff
@@ -2156,6 +2159,308 @@ var _ = Describe("AppHandler", func() {
...
+				Expect(appRepo.SetAppDesiredStateCallCount()).To(Equal(2))
+
+				_, _, appDesiredStateMessage := appRepo.SetAppDesiredStateArgsForCall(0)
+				Expect(appDesiredStateMessage.DesiredState).To(Equal("STOPPED"))
+
+				_, _, appDesiredStateMessage = appRepo.SetAppDesiredStateArgsForCall(1)
+				Expect(appDesiredStateMessage.DesiredState).To(Equal("STARTED"))
+			})
+
+			It("calls WatchForPodsTermination to wait before staring the app", func() {
```
Comment (tcdowney) [inline api/apis/app_handler_test.go:2209]: Nit: Typo on "staring"

**Review by tcdowney — APPROVED** [review]
A couple minor typos, but otherwise LGTM.

**Pull request merged**